### PR TITLE
Add PocketBase

### PR DESCRIPTION
**Pull request by sharknoon**
# App Submission

### App name
PocketBase

### 256x256 SVG icon
pocketbase

### Gallery images
Screenshots from the [official demo](https://pocketbase.io/demo/)

### I have tested my app on:
- [ ] umbrelOS on a Raspberry Pi
- [ ] umbrelOS on an Umbrel Home
- [x] umbrelOS on Linux VM

### CODE REVIEW

File: pocketbase/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      PROXY_AUTH_ADD: "false"
+  
+  app:
+    image: ghcr.io/muchobien/pocketbase:0.22.11@sha256:04dd9850554a24f82cfee060ff394179baaecc0f97e7721d9dd06818d26fd671
+    restart: on-failure
+    stop_grace_period: 1m
+    volumes:
+      - ${APP_DATA_DIR}/app/data:/pb_data
+      - ${APP_DATA_DIR}/app/public:/pb_public
+      - ${APP_DATA_DIR}/app/hooks:/pb_hooks
```
Comment (highghlow) [inline pocketbase/docker-compose.yml:17]:
```suggestion
      - ${APP_DATA_DIR}/data/data:/pb_data
      - ${APP_DATA_DIR}/data/public:/pb_public
      - ${APP_DATA_DIR}/data/hooks:/pb_hooks
```

App data is usually stored in ${APP_DATA_DIR}/data

**Comment by sharknoon** [comment]
Thank you @highghlow for reviewing. I have updated the `docker-compose.yml` accordingly 😃

File: pocketbase/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      PROXY_AUTH_ADD: "false"
+  
+  app:
+    image: ghcr.io/muchobien/pocketbase:0.22.11@sha256:04dd9850554a24f82cfee060ff394179baaecc0f97e7721d9dd06818d26fd671
+    restart: on-failure
+    stop_grace_period: 1m
+    volumes:
+      - ${APP_DATA_DIR}/data/data:/pb_data
+      - ${APP_DATA_DIR}/data/public:/pb_public
+      - ${APP_DATA_DIR}/data/hooks:/pb_hooks
```
Comment (nmfretz) [inline pocketbase/docker-compose.yml:17]: Can you please add these 3 host paths with a `.gitkeep` in each as described here (https://github.com/getumbrel/umbrel-apps/pull/1083#discussion_r1609144592)?

File: pocketbase/umbrel-app.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  
+  💾 File storage
+  
+  Sanely store files locally or in a S3 storage. Easily attach media to your database records and generate thumbs on the fly.
+  
+  
+  🧩 Extendable
+  
+  Use as a standalone app OR as a framework, that you can extend via Go and JavaScript hooks to create your own custom portable backend.
+developer: Gani
```
Comment (nmfretz) [inline pocketbase/umbrel-app.yml:30]: Let's use `PocketBase` here as the developer. The github repo is under the PocketBase organization and the website shows this:

image

File: pocketbase/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: pocketbase_app_1
+      APP_PORT: 8090
+      PROXY_AUTH_ADD: "false"
+  
+  app:
```
Comment (nmfretz) [inline pocketbase/docker-compose.yml:10]: Can you please try out the app with this container running under `user: "1000:1000"`. Right now the container is running as `root`, but if we can reduce ownership/permissions to the `umbrel` user then this is ideal from an app sandbox point of view.

**Review by nmfretz — COMMENTED** [review]
Excellent, thanks for submitting PocketBase @sharknoon. We'll get gallery images made up.

I've left some minor suggestions below.

**Comment by sharknoon** [comment]
Hi @nmfretz,
thank you for reviewing the PR. Also thank you for the email you send me. I feel honored and will answer the email shortly.
I have made the necessary changes.
Kind regards
Josua aka Sharknoon

File: pocketbase/docker-compose.yml (hunk 4 above)
Comment (sharknoon) [reply pocketbase/docker-compose.yml:10]: I have tried using `user: 1000:1000`, but unfortunately I was not able to create collections within pocketbase. Then I have removed the user... line and it worked. To double check I have added it back and the error unfortunately came back

Comment (nmfretz) [reply pocketbase/docker-compose.yml:10]: Thanks for testing this @sharknoon 🤝. I have added a comment to document that pocketbase needs to be run with root permissions https://github.com/getumbrel/umbrel-apps/pull/1082/commits/ed35d230a4dde9dbba6ed66957dc97727d122469

**Comment by nmfretz** [comment]
@sharknoon gallery assets are here: https://github.com/getumbrel/umbrel-apps-gallery/tree/master/pocketbase

Thanks again for submitting PocketBase! I have added some minor commits to finalize. Going live to the app store now 🎉

image

**Pull request merged**